src/librustc_resolve/lib.rs

```diff
@@ -3157,16 +3157,46 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
                                 Ok(ref snippet) if snippet == "{" => true,
                                 _ => false,
                             };
-                            if let (PathSource::Expr(None), true) = (source, followed_by_brace) {
-                                err.span_label(
-                                    span,
-                                    format!("did you mean `({} {{ /* fields */ }})`?", path_str),
-                                );
-                            } else {
-                                err.span_label(
-                                    span,
-                                    format!("did you mean `{} {{ /* fields */ }}`?", path_str),
-                                );
+                            match source {
+                                PathSource::Expr(Some(parent)) => {
+                                    match parent.node {
+                                        ExprKind::MethodCall(ref path_assignment, _)  => {
+                                            err.span_suggestion_with_applicability(
+                                                parent.span,
```

Comment:
Instead of `parent.span` use `source_map.start_point(parent.span).to(path_assignment.span)`. This way the span will point at `Span.new` instead of `Span.new()`.

Reply:
then you can get the span through its `ident` field.